analytic_distribution_filter/models/analytic_account_line.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import Command, api, fields, models
+
+
+class AnalyticAccountLine(models.Model):
+    _name = "analytic.account.line"
+
+    account_id = fields.Many2one("account.analytic.account", ondelete="cascade")
+    plan_id = fields.Many2one("account.analytic.plan", required=True)
+    account_ids = fields.Many2many(
+        "account.analytic.account", string="Related Accounts", required=True
+    )
+
+    @api.onchange("plan_id")
+    def onchange_plan_id(self):
+        self.account_ids = False
+
+    def _search_reciprocal(self, account_id, plan_id):
+        return self.search(
+            [
+                ("account_id", "=", account_id),
+                ("plan_id", "=", plan_id),
+            ],
+            limit=1,
+        )
+
+    def _update_reciprocal_relations(self, old_account_ids=None):
+        for line in self:
+            current_ids = set(line.account_ids.ids)
+            old_ids = set(old_account_ids or [])
+            added_ids = current_ids - old_ids
+            removed_ids = old_ids - current_ids
+            for account_id in added_ids:
+                reciprocal = self._search_reciprocal(
+                    account_id, line.account_id.plan_id.id
+                )
+                if reciprocal:
+                    if line.account_id.id not in reciprocal.account_ids.ids:
+                        reciprocal.account_ids = [Command.link(line.account_id.id)]
+                else:
+                    self.create(
+                        {
+                            "account_id": account_id,
+                            "plan_id": line.account_id.plan_id.id,
+                            "account_ids": [Command.set([line.account_id.id])],
+                        }
+                    )
```

Comment:
To improve performance.
```suggestion
            existing_recip_lines = self._get_reciprocal_lines(added_ids, line.account_id.plan_id.id)
            missing_ids = added_ids - set(existing_recip_lines.account_id.ids)
            if missing_ids:
                self.create(
                    [
                        {
                            "account_id": aid,
                            "plan_id": line.account_id.plan_id.id,
                            "related_account_ids": [Command.set([line.account_id.id])],
                        }
                        for aid in missing_ids
                    ]
                )
            if existing_recip_lines:
                    existing.write({"related_account_ids": [Command.link(line.account_id.id)]})
```